src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -3534,21 +3533,22 @@ private static String getValueString(int signum, String intString, int scale) {
         StringBuilder buf;
         int insertionPoint = intString.length() - scale;
         if (insertionPoint == 0) {  /* Point goes just before intVal */
-            return (signum<0 ? "-0." : "0.") + intString;
+            return (signum < 0 ? "-0." : "0.").concat(intString);
         } else if (insertionPoint > 0) { /* Point goes inside intVal */
-            buf = new StringBuilder(intString);
-            buf.insert(insertionPoint, '.');
+            buf = new StringBuilder();
             if (signum < 0)
-                buf.insert(0, '-');
+                buf.append('-');
+            buf.append(intString)
+                    .insert(insertionPoint + (signum < 0 ? 1 : 0), '.');
```

Comment:
Instead of the insert, could do an append of the  prefix, then dot, then the suffix.

Reply:
```java
buf.append(intString, 0, insertionPoint)
   .append('.')
   .append(intString, insertionPoint, intString.length());
```

This is another way to write it, but the performance will be reduced.